### PR TITLE
net/cord doc: Usability fixes

### PR DESCRIPTION
**Pull request by chrysn**
### Contribution description

The CoRD documentation had a few shortcomings, some of which doxygen already warned about, some from general usability, some because the text I added in #16113 was a bit too concise.

### Testing procedure

* Documentation builds without errors (even though some exclude patterns from #16779 were removed)
* All CoRD configuration macros are shown and searchable (for example, CONFIG_CORD_EP used to be shown but could not be found through the search;  CONFIG_CORD_EXTRAARGS was not shown before at all).

### CODE REVIEW

**Comment by chrysn** [comment]
As this removes doxygen exclusions from 16779, to avoid holding up the upcoming CI changes of https://github.com/RIOT-OS/riotdocker/pull/104 with an accidental regression, this should wait for the new riotdocker images to run the checks against.

File: sys/include/net/cord/config.h [hunk 1]
```diff
@@ -7,8 +7,8 @@
  */
 
 /**
- * @defgroup    net_cord_config CoRE RD Endpoint and Lookup Client Configuration
- * @ingroup     net_cord
+ * @defgroup    net_cord_config CoRE RD Configuration
+ * @ingroup     net_cord config
```
Comment (miri64) [inline sys/include/net/cord/config.h:11]:
```suggestion
 * @ingroup     net_cord
 * @ingroup     config
```

At least that's how it is done elsewhere.

**Comment by fjmolinas** [comment]
It doesn't seem to need the CI update anymore no?

**Comment by chrysn** [comment]
Indeed; flag removed and miri's suggestions pulled in.

As there were no more review items other than the one that is simple to follow, but warnings about merge conflicts, I'm taking the liberty to rebase.

**Comment by miri64** [comment]
Thx!

**Pull request merged**